src/app/notifications/notification-action.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * An action for a button, dropdown, etc:
+ *
+ * id - Optional unique Id for the filter field, useful for comparisons
+ * isDisabled - set to true to disable the action
+ * isSeparator - set to true if this is a placehodler for a separator rather than an action
+ * name - The name of the action, displayed on the button
+ * title - Optional title, used for the tooltip
+ */
+export class NotificationAction {
+  id?: string;
+  isDisabled?: boolean = false;
+  isSeparator?: boolean = false;
+  name: string;
+  title?: string;
+  callback: NotificationActionCallback;
```

Comment:
Notification also doesn't fit very well with the notificationservice, as you can't call notificationservice.message(notification) and it has properties that notification service will override.

Also I think we need the model classes outside of a widgets library so that they can be properly decoupled for users.